testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/active_request/ActiveRequestTrackerHandlerTest.java

```diff
@@ -0,0 +1,139 @@
+            + "         xsi:schemaLocation=\"http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd\"\n"
+            + "         metadata-complete=\"false\" version=\"3.0\">\n"
+            + "    <servlet-mapping>\n"
+            + "        <servlet-name>javax.ws.rs.core.Application</servlet-name>\n"
+            + "        <url-pattern>/*</url-pattern>\n"
+            + "    </servlet-mapping>"
+            + "</web-app>";
+
+    @Deployment
+    public static WebArchive deploy() {
+        return ShrinkWrap.create(WebArchive.class, "active-request-tracking.war")
+                .addClasses(DelayServlet.class);
+    }
+
+    @ArquillianResource
+    protected URL url;
+
+    @Test
+    public void testActiveRequest() throws Exception {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            ManagementClient managementClient = new ManagementClient(TestSuiteEnvironment.getModelControllerClient(),
+                    TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getServerPort(), "remote+http");
+
+
+            new Thread(() -> makeRequest(httpClient)).start();
+            // This test is somewhat time-sensitive in that an in-flight request is needed to verify that the handler
+            // is correctly tracking and returning active requests. To that end, the test submits a request in a
+            // Thread (above) to let the request be made in a background thread while the main thread attempts to
+            // get the active request information. There is a short sleep here to force/encourage the JVM to allow the
+            // background thread to start to initiate the request. That thread has a short delay in it as well (see
+            // servlet) to give this thread time to retrieve the data.
+            Thread.sleep(500);
+
+            ModelNode response = managementClient.getControllerClient().execute(
+                    Operations.createOperation("list-active-requests",
+                            Operations.createAddress(
+                                    "subsystem", "undertow",
+                                    "server", "default-server",
+                                    "host", "default-host",
+                                "setting", "active-request-tracker")
+            ));
+
+            assertEquals(new ModelNode("success"), response.get("outcome"));
+            ModelNode result = response.get("result");
+            assertEquals(ModelType.LIST, result.getType());
+            assertEquals(1, result.asList().size());
+        }
+    }
+
+    private void makeRequest(CloseableHttpClient httpClient) {
+        try {
+            httpClient.execute(new HttpGet(url.toExternalForm() + "test"));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static class ActiveRequestTrackingSetupTask implements ServerSetupTask {
```

Comment:
Pro tip: leverage CLIServerSetupTask for this.
e.g. https://github.com/wildfly/wildfly/blob/main/testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/remote/InfinispanServerSetupTask.java